vault/resource_okta_auth_backend_test.go

```diff
@@ -0,0 +1,255 @@
+package vault
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/hashicorp/vault/api"
+	"testing"
+	"time"
+)
+
+func TestOktaAuthBackend(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testAccPreCheck(t) },
+		CheckDestroy: testOktaAuthBackend_Destroyed,
+		Steps: []resource.TestStep{
+			{
+				Config: initialOktaAuthConfig,
+				Check:  testOktaAuthBackend_InitialCheck,
+			},
+			{
+				Config: updatedOktaAuthConfig,
+				Check:  testOktaAuthBackend_UpdatedCheck,
+			},
+		},
+	})
+}
+
+var initialOktaAuthConfig = `
+resource "vault_okta_auth_backend" "test" {
+    description = "Testing the Terraform okta auth backend"
+    organization = "example"
+    token = "this must be kept secret"
+    ttl = "1h"
+    group {
+        group_name = "dummy"
```

Comment:
We like making any test data that requires uniqueness randomized, so multiple tests can run at once and so if a test doesn't clean up properly after itself, it doesn't break integration tests until someone goes in and cleans it up. You can find some helpers for randomising these names in the acctest package: https://godoc.org/github.com/hashicorp/terraform/helper/acctest

Generally how this is done is by making the config a function, passing in randomised or environment-determined variables, injecting them with `fmt.Sprintf`, and initializing all these variables in the `TestAcc*` functions. You can see a few examples in the tests for the AWS auth backend.